frame/multisig/src/lib.rs

```diff
@@ -485,12 +509,126 @@ pub mod pallet {
 			debug_assert!(err_amount.is_zero());
 			<Multisigs<T>>::remove(&id, &call_hash);
 
-			Self::deposit_event(Event::MultisigCancelled {
-				cancelling: who,
-				timepoint,
-				multisig: id,
-				call_hash,
-			});
+			Self::deposit_event(Event::MultisigCancelled { timepoint, multisig: id, call_hash });
+			Ok(())
+		}
+
+		/// Register approval for a dispatch to be made from a deterministic composite account if
+		/// approved by a total of `threshold - 1` of `other_signatories`.
+		///
+		/// Payment: `DepositBase` will be reserved if this is the first approval, plus
+		/// `threshold` times `DepositFactor`. It is returned once this dispatch happens or
+		/// is cancelled.
+		///
+		/// The dispatch origin for this call must be _Signed_.
+		///
+		/// - `threshold`: The total number of approvals for this dispatch before it is executed.
+		/// - `other_signatories`: The accounts (other than the sender) who can approve this
+		/// dispatch. May not be empty.
+		/// - `maybe_timepoint`: If this is the first approval, then this must be `None`. If it is
+		/// not the first approval, then it must be `Some`, with the timepoint (block number and
+		/// transaction index) of the first approval transaction.
+		/// - `call_hash`: The hash of the call to be executed.
+		/// - `expiry`: An optional expiry of the multisig operation defined in block number.
+		///
+		/// NOTE: This extrinsic is only intended to be used for creating a multisig with an expiry.
+		/// It should not be used for approving existing multisigs.
+		///
+		/// ## Complexity
+		/// - `O(S)`.
+		/// - Up to one balance-reserve or unreserve operation.
+		/// - One passthrough operation, one insert, both `O(S)` where `S` is the number of
+		///   signatories. `S` is capped by `MaxSignatories`, with weight being proportional.
+		/// - One encode & hash, both of complexity `O(S)`.
+		/// - Up to one binary search and insert (`O(logS + S)`).
+		/// - I/O: 1 read `O(S)`, up to 1 mutate `O(S)`. Up to one remove.
+		/// - One event.
+		/// - Storage: inserts one item, value size bounded by `MaxSignatories`, with a deposit
+		///   taken for its lifetime of `DepositBase + threshold * DepositFactor`.
+		#[pallet::call_index(4)]
+		#[pallet::weight({
+			let s = other_signatories.len() as u32;
+
+			// The weight is the same as for `approve_as_multi` but there is an
+			// additional write since we are writing into `MultisigExpiries`. 
+			T::WeightInfo::approve_as_multi_create(s)
+				.max(T::WeightInfo::approve_as_multi_approve(s))
+				.saturating_add(*max_weight)
+				.saturating_add(T::DbWeight::get().writes(1))
+		})]
+		pub fn create_multisig_with_expiry(
```

Comment:
What do you think about
- Rename existing `approve_as_multi` to `approve_as_multi_legacy` but keep it in the same callindex (which I believe should make it non-breaking? need someone to confirm)
- Name this method `approve_as_multi`
- Make the blocknumber arg here an Option

Otherwise we have duplicate methods, one that defaults to a `None` expiry and one that enforces a `Some` expiry. These should be combined into a single method.

Reply:
I overall agree with your suggestions.

> Rename existing approve_as_multi to approve_as_multi_legacy but keep it in the same callindex (which I believe should make it non-breaking? need someone to confirm)

I am pretty sure that renaming an existing extrinsic without changing the call index will not be a breaking change.

Reply:
Confirmed this should be fine: https://substrate.stackexchange.com/questions/246/does-renaming-an-extrinsic-require-a-transaction-version-bump